CMP1903_A1_2324/Testing.cs

```diff
@@ -1,20 +1,23 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Diagnostics;
 
-namespace CMP1903_A1_2324
+namespace CMP1903_A1_2324 //defines the namespace
 {
-    internal class Testing
+    internal class Testing //declares the class called Testing
     {
-        /*
-         * This class should test the Game and the Die class.
-         * Create a Game object, call the methods and compare their output to expected output.
-         * Create a Die object and call its method.
-         * Use debug.assert() to make the comparisons and tests.
-         */
+        public void TestGameAndDie(Game game) //declares the TestGameAndDie method and takes a Game object as a parameter
+        {
+            //tests the Game class            
+            int total = game.GetTotal(); //calls the gettotal method to get the total of the dice rolls
+            //asserts that the total of the three dice rolls is between 3 and 18. If not an error message is produced
+            Debug.Assert(total >= 3 && total <= 18, "Total of three dice rolls is not as between 3 and 18.");
 
-        //Method
+            //tests the Die class
+            Die die = new Die(); //creates new instance of the die class to test if rolls are between 1 and 6
+            int rolledValue = die.Roll(); //calls the roll method and stores the value of the roll
+            //asserts that the rolled value is be between 1 and 6. if not an error message is produced
+            Debug.Assert(rolledValue >= 1 && rolledValue <= 6, "Rolled value is not between 1 and 6.");
+        }
     }
 }
+
```

Comment:
Ultimately, your code is well structured and you've shown a good understanding of instantiation and encapsulation.  To improve, you can consider making more detailed comments and include error handling.